server/server_test.go

```diff
@@ -1682,6 +1682,112 @@ func TestDataPutV1IfNoneMatch(t *testing.T) {
 	}
 }
 
+func mustGZIPPayload(payload []byte) []byte {
+	// Compress JSON payload with gzip
+	var compressedPayload bytes.Buffer
+	gz := gzip.NewWriter(&compressedPayload)
+	if _, err := gz.Write(payload); err != nil {
+		panic(fmt.Errorf("Error closing gzip writer: %w", err))
```

Comment:
[nit]
```suggestion
		panic(fmt.Errorf("Error writing to gzip writer: %w", err))
```

Reply:
Good suggestion! I've integrated that into my latest changes.